src/util.h

```diff
@@ -120,9 +126,10 @@ void DumpBacktrace(FILE* fp);
 #define CHECK(expr)                                                           \
   do {                                                                        \
```

Comment:
If you're here, would you consider de-crufting this? i.e.:
```
    if (!(expr)) {                                                             \
      node::Assert({ __FILE__, __LINE__, #expr, PRETTY_FUNCTION_NAME });       \
    }                                                                          \
```

Reply:
@refack At least locally (gcc-7 on Ubuntu 18.04), that would undo the purpose of this (what the comment is referring to): It would put the object construction in-line into the code instead of ending up as a simple pointer load from the read-only data section.

Reply:
So how about:
```c++
    if (!(expr)) {                                                             \
      static constexpr args{ __FILE__, __LINE__, #expr, PRETTY_FUNCTION_NAME } \
      node::Assert(args);                                                      \
    }
```

Reply:
@refack You are talking about `const` → `constexpr`, right? That has the same result for me, and I don’t see it as being clearer about the intention here (having an explicit, hardcoded object in the read-only data section).

I’ll make the comment more explicit about what the idea here is.